unit_test/entities/test_production_cluster.py

```diff
@@ -0,0 +1,124 @@
+#  Copyright (c) 2023. Deltares & TNO
+#
+#  This program is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+"""Test Junction entities."""
+import unittest
+
+import pandapipes as pp
+import pytest
+
+from simulator_core.entities.assets import Junction, ProductionCluster
+from simulator_core.entities.assets.asset_defaults import (
+    PROPERTY_HEAT_DEMAND,
+    PROPERTY_TEMPERATURE_RETURN,
+    PROPERTY_TEMPERATURE_SUPPLY,
+)
+from simulator_core.entities.assets.utils import heat_demand_and_temperature_to_mass_flow
+
+
+class ProductionClusterTest(unittest.TestCase):
+    """Testcase for ProductionCluster class."""
+
+    def setUp(self):
+        """Set up test case."""
+        # Create empty pandapipes network
+        self.network = pp.create_empty_network(fluid="water")
+        # Create two junctions
+        self.from_junction = Junction(name="from_junction", pandapipes_net=self.network)
+        self.to_junction = Junction(name="to_junction", pandapipes_net=self.network)
+        # Create a production cluster object
+        self.production_cluster = ProductionCluster(
+            asset_name="production_cluster",
+            asset_id="production_cluster_id",
+            pandapipe_net=self.network,
+        )
+        self.production_cluster.connect_junctions(
+            from_junction=self.from_junction, to_junction=self.to_junction
+        )
+
+    def test_production_cluster_create(self):
+        """Evaluate the creation of a production_cluster object."""
+        # Arrange
+
+        # Act
+        self.production_cluster.create()
+
+        # Assert
+        assert isinstance(self.production_cluster, ProductionCluster)
+        assert self.production_cluster.name == "production_cluster"
+        assert self.production_cluster.asset_id == "production_cluster_id"
+        assert any(self.network.flow_control.name == "flow_control_production_cluster")
```

Comment:
You also check whether a components has been made in the network: "assert any(self.network.flow_control.name == "flow_control_production_cluster")". Other settings are not set because I only create the asset.

Reply:
I do not necessarily see how adding all stuff will improve the quality of the test.